source/codegen/client_helpers.py

```diff
@@ -20,8 +21,7 @@ class ParamMechanism(Enum):
   ])
 
 
-def create_params(func: dict) -> List[str]:
-  client_params = get_client_parameters(func)
+def to_parameter_list(client_params: List[ClientParam]) -> List[str]:
```

Comment:
Changing methods to take in `List[ClientParam]` avoids needing to pass enums into every method on the API. We only need the enums to generate the `ClientParam`s.